mat64/vector.go

```diff
@@ -357,6 +360,61 @@ func (v *Vector) MulVec(a Matrix, b *Vector) {
 	}
 }
 
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+//
+// Vector is little-endian encoded as follows:
+//   0 -  8  number of elements     (int64)
+//   8 - ..  vector's data elements (float64)
+func (v Vector) MarshalBinary() ([]byte, error) {
+	buf := bytes.NewBuffer(make([]byte, 0, sizeInt64+v.n*sizeFloat64))
+	err := binary.Write(buf, defaultEndian, int64(v.n))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, val := range v.mat.Data {
```

Comment:
it could be surprising from outside and break invariants of users' code. Consider

```go
package my

type View struct {
    vec mat64.Vector
    inc int
}

func (view View) MarshalBinary(...) {...}
func (view *View) UnmarshalBinary(...) {...}
```

that could potentially break users' marshalling code.
we could of course document it, but it seems to me like an unnecessary hurdle (and it would just move the problem to someone else having to also properly document this gotcha)